File: python-examples/native-crawler/utils/content.py
from playwright.async_api import Page

from intuned_browser import extract_markdown
from intuned_browser.ai import extract_structured_data


async def extract_page_content(
    page: Page,
    schema: dict | None = None,
) -> dict:
    if schema:
        return await extract_structured_data(source=page, data_schema=schema)

    title = await page.title()
    markdown = await extract_markdown(page)

    return {
        "title": title,
        "markdown": markdown,
        "markdown_length": len(markdown) if markdown else 0,
    }

File: typescript-examples/native-crawler/utils/content.ts
import { Page } from "playwright";
import { extractMarkdown } from "@intuned/browser";
import { extractStructuredData } from "@intuned/browser/ai";

export async function extractPageContent(
  page: Page,
  schema?: any
): Promise<any> {
  if (schema) {
    return await extractStructuredData({
      source: page,
      dataSchema: schema,
    });
  }

  const title = await page.title();
  const markdown = await extractMarkdown({ source: page });

  return {
    title,
    markdown,
    markdown_length: markdown?.length || 0,
  };
}
